src/legacy/core_plugins/vis_default_editor/public/components/controls/time_interval.tsx

```diff
@@ -56,15 +81,46 @@ function TimeIntervalParamEditor({
   let selectedOptions: ComboBoxOption[] = [];
   let definedOption: ComboBoxOption | undefined;
   let isValid = false;
+  let invalidEsMessage = '';
+  const errors = [];
   if (value) {
     definedOption = find(options, { key: value });
     selectedOptions = definedOption ? [definedOption] : [{ label: value, key: 'custom' }];
-    isValid = !!(definedOption || isValidInterval(value, timeBase));
+
+    // we check if Elasticsearch interval is valid ES interval to show a user appropriate error message
+    // we don't check if there is timeBase
+    const isValidIntervalValue = timeBase ? true : isValidEsInterval(value);
+
+    if (!isValidIntervalValue) {
+      invalidEsMessage = getInvalidEsMessage(value);
+    }
+
+    isValid = !!definedOption || (isValidIntervalValue && isValidInterval(value, timeBase));
```

Comment:
Seems when you have a `definedOption` from a list of options, you don't need further checks.
The validation of interval in this component grew a lot (become near 40 lines of code), I would recommend to restructure it a bit or move into a helper function to clean up the component body.

Reply:
I updated the PR, please take a look.